lambda-search/src/embed.rs

```diff
@@ -0,0 +1,197 @@
+use anyhow::{anyhow, bail, Result};
+use reqwest::{
+    header::{HeaderMap, HeaderName},
+    Client, RequestBuilder, Url,
+};
+use serde_json::Value as JsonValue;
+
+#[derive(Clone)]
+enum RequestType {
+    Get {
+        query_key: &'static str,
+    },
+    Post {
+        prefix: &'static str,
+        suffix: &'static str,
+    },
+}
+
+impl RequestType {
+    fn request(&self, client: &Client, url: Url, text: &str) -> RequestBuilder {
+        match self {
+            RequestType::Get { query_key } => {
+                let mut url = url;
+                url.query_pairs_mut().append_pair(query_key, text);
+                client.get(url)
+            }
+            RequestType::Post { prefix, suffix } => {
+                client.post(url).body(format!("{prefix}{text}{suffix}"))
+            }
+        }
+    }
+}
+
+pub struct Embedding {
+    url: Url,
+    request_type: RequestType,
+    headers: HeaderMap,
+    result_keys: &'static [&'static str],
+    multi: bool,
+}
+
+impl Embedding {
+    pub fn from_env() -> Self {
+        match std::env::var("EMBEDDING").as_deref() {
+            Ok("COHERE") => Self::cohere(),
+            Ok("MIGHTY") => Self::mighty(
+                &std::env::var("MIGHTY_URI").expect("needs MIGHTY_URI env variable set"),
+            ),
+            Ok("OPEN_AI") => Self::open_ai(),
+            _ => panic!("EMBEDDING must be one of COHERE, MIGHTY or OPENAI"),
+        }
+    }
+
+    // create a mighty embedding
```

Comment:
I guess we should use `///`, in other places as well, to make it a doc comment.

```suggestion
    /// Create a mighty embedding
```